azkaban-common/src/main/java/azkaban/executor/ExecutorApiClient.java

```diff
@@ -196,7 +206,8 @@ protected CloseableHttpClient createHttpsClient() {
    * @return the response object type of which is specified by user.
    * @throws UnsupportedEncodingException, IOException.
    */
-  public String httpsPost(final URI uri, final List<Pair<String, String>> params)
+  public String httpsPost(final URI uri,
+      final Optional<Integer> httpTimeout, final List<Pair<String, String>> params)
```

Comment:
Please consider passing a regular Integer (as opposed to Optional<Integer>) with value -1 when there is no timeout.